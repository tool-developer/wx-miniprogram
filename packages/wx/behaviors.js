module.exports = require('./miniprogram_dist/behaviors');