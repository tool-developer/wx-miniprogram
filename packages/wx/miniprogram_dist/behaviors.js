import wxapp from './index.js';

var behaviors = Behavior({
  behaviors: [],
  properties: {},
  data: {},
  methods: { ...wxapp
  },

  definitionFilter(defFields) {
    //
    defFields.methods.__getEvents = () => defFields.events || {};
  },

  created() {
    //
    Object.assign(this, wxapp, {
      data: this.data
    }); // 扩展wxapp方法到this
    // 扩展events

    this.events = wxapp.extend(this.__getEvents(), this.events);
  }

});

export default behaviors;
