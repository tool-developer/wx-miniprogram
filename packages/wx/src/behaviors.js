import wxapp from './index';

export default Behavior({
  behaviors:[],
  properties:{},
  data:{},
  methods:{
    ...wxapp
  },
  definitionFilter(defFields){
    //
    defFields.methods.__getEvents = () => defFields.events
  },
  created(){
    //
    Object.assign(this,wxapp);
    // 扩展events
    this.events = wxapp.extend(this.__getEvents(),this.events);
  }
})