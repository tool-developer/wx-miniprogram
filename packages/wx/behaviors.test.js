import behaviors from './behaviors';

test('wx app behaviors test', () => {
  //
  global.Behavior = behavior=>behavior;
  // 固定index.test.js.snap中该字段时间
  behaviors.methods.__SET_DATA_FREQUENCY_TIMESTAMP = 1625559145942;
  //
  expect(behaviors).toMatchSnapshot();
});