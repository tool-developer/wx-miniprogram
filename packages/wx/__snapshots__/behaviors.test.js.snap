// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wx app behaviors test 1`] = `
Object {
  "behaviors": Array [],
  "created": [Function],
  "data": Object {},
  "definitionFilter": [Function],
  "methods": Object {
    "$set": [Function],
    "PageNamePathKV": Object {
      "webview": "pages/webview/index",
    },
    "SET_DATA_FREQUENCY": 100,
    "TabBarList": Array [],
    "__SET_DATA_CACHE": Object {},
    "__SET_DATA_FREQUENCY_TIMESTAMP": 1625559145942,
    "authorize": [Function],
    "back": [Function],
    "bindEvent": [Function],
    "bindTelEvent": [Function],
    "data": Object {},
    "dataset": [Function],
    "events": Object {},
    "extend": [Function],
    "getWxCurrentOptions": [Function],
    "getWxCurrentPage": [Function],
    "getWxSetting": [Function],
    "getWxUserInfoAuthorize": [Function],
    "go": [Function],
    "hideLoading": [Function],
    "hideToast": [Function],
    "i18n": Object {
      "authorize.cancel": "取消授权",
      "authorize.content": "{appName}需要您授权{scopeName}，点击确定前去设置",
      "authorize.loading": "正在授权...",
      "authorize.success": "已授权",
      "authorize.title": "授权提示",
    },
    "i18nHandle": [Function],
    "navigateTo": [Function],
    "pageTo": [Function],
    "qs": [Function],
    "reLaunch": [Function],
    "reLaunchTo": [Function],
    "redirectTo": [Function],
    "scrollToEl": [Function],
    "showLoading": [Function],
    "showModal": [Function],
    "showToast": [Function],
    "switchTab": [Function],
    "switchTo": [Function],
    "version": "1.0.0",
  },
  "properties": Object {},
}
`;
